clients/general/client.go

```diff
@@ -34,38 +33,30 @@ type GeneralClient interface {
 }
 
 type generalRestClient struct {
-	url      string
-	endpoint interfaces.Endpointer
+	urlClient interfaces.URLClient
 }
 
 // NewGeneralClient creates an instance of GeneralClient
 func NewGeneralClient(params types.EndpointParams, m interfaces.Endpointer) GeneralClient {
-	gc := generalRestClient{endpoint: m}
-	gc.init(params)
-	return &gc
+	return &generalRestClient{urlClient: urlclient.New(params, m)}
 }
 
-func (gc *generalRestClient) init(params types.EndpointParams) {
-	if params.UseRegistry {
-		go func(ch chan string) {
-			for {
-				select {
-				case url := <-ch:
-					gc.url = url
-				}
-			}
-		}(gc.endpoint.Monitor(params))
-	} else {
-		gc.url = params.Url
+func (gc *generalRestClient) FetchConfiguration(ctx context.Context) (string, error) {
+	url, err := gc.urlClient.Prefix()
```

Comment:
Variable naming in prior PRs is `urlPrefix`; rename `url` here for consistency.  Applies to any/all calls to Prefix().